controllers/application_controller.go

```diff
@@ -165,10 +187,85 @@ func (r *ApplicationReconciler) reconcileHRStatus(application *v1.Application, c
 		v1.AppErrorStatus(application, "updated Helm Release status not available")
 		return errors.New("HelmRelease status is not current")
 	}
-	application.Status.Conditions = hr.GetConditions()
+	for _, condition := range hr.GetConditions() {
+		apimeta.SetStatusCondition(&application.Status.Conditions, condition)
+	}
 	return nil
 }
 
+func (r *ApplicationReconciler) reconcilePodStatus(ctx context.Context, application *v1.Application) error {
+	// Analyze the status of the pods under the HR, if any
+	log.Info("Reconciling pod status", "application", application.Name, "namespace", application.Namespace)
+	// XXX: This is for deployments. We need to make it clear that right now, this feature is "best effort".
+	var deploymentList appsv1.DeploymentList
+	if err := r.List(ctx, &deploymentList, &client.ListOptions{Namespace: application.Namespace}); err != nil {
+		return err
+	}
+	// Try to retrieve the deployment
+	var releaseDeployment *appsv1.Deployment
+	for _, deployment := range deploymentList.Items {
+		if deployment.GetAnnotations()[AnnotationHelmReleaseName] == application.Name && deployment.GetAnnotations()[AnnotationHelmReleaseNamespace] == application.Namespace {
+			releaseDeployment = &deployment
+			break
+		}
+	}
+	if releaseDeployment == nil {
+		// We didn't find a release deployment, so no pod status.
```

Comment:
This isn't really a problem, because the HelmRelease has control of the `Ready` condition which dictates the true "health" of the application. In other words, as long as the HelmRelease is in a non-ready state, the Application will not be either.

Reply:
but you would be providing an incorrect analysis condition.

Reply:
You're right. 

Resolved by ca5f866